### PR TITLE
#161368021 admin can create new category

### PR DESCRIPTION
**Pull request by onwuvic**
#### What does this PR do?
An admin can add a new category

#### Description of Task to be completed?
- Add admin permission to the create category routes
- write a test to test the condition

#### How should this be manually tested?
- Sign up as an admin, then create a new category

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
Story type: feature
Story title: Admin should be able to create new categories
Story ID: #161368021

#### Screenshots (if appropriate)
N/A
#### Questions:```
N/A

### CODE REVIEW

File: server/controllers/reportController.js [hunk 1]
```diff
@@ -19,7 +19,6 @@ export default class ReportController {
 
     return Article.findOne({
       where: {
-        userId: data.userId,
```
Comment (onwuvic) [inline server/controllers/reportController.js:22]: The authenticated user ID is not needed to find a reported article

File: server/migrations/20181016125446-create-report.js [hunk 2]
```diff
@@ -10,19 +10,19 @@ module.exports = {
     userId: {
       allowNull: false,
       type: Sequelize.INTEGER,
+      onDelete: 'CASCADE',
```
Comment (onwuvic) [inline server/migrations/20181016125446-create-report.js:13]: This allows a reported article to be deleted and also delete its reference on the report table

File: server/routes/api/article.js [hunk 3]
```diff
@@ -16,9 +17,9 @@ const router = express.Router();
 router.post('/articles', checkToken, upload.single('imgUrl'), ArticleValidations.validateCreateArticle, ArticleController.createArticle);
 router.get('/articles/feed', ArticleController.getAllArticles);
 router.get('/articles', checkToken, ArticleController.getUserArticles);
-router.get('/categories', checkToken, CategoryController.getAllCategories);
-router.post('/categories', checkToken, CategoryController.createCategory);
```
Comment (onwuvic) [inline server/routes/api/article.js:21]: Remove authentication from this route. They are not needed. A user who is not login should be able to read articles by categories.

File: server/tests/categories.test.js [hunk 4]
```diff
@@ -1,33 +1,36 @@
 /* eslint-disable prefer-destructuring */
-import faker from 'faker';
 import chai, { expect } from 'chai';
 import chaiHttp from 'chai-http';
 import app from '../index';
 
 chai.use(chaiHttp);
-let token = '';
-
-const user = {
```
Comment (onwuvic) [inline server/tests/categories.test.js:10]: I used seeder data instead of creating another user here to avoid duplication.

**Review by s-akinrele — APPROVED** [review]

**Pull request merged**